library/aes.c

```diff
@@ -979,7 +995,10 @@ int mbedtls_aes_crypt_cfb128( mbedtls_aes_context *ctx,
 
     *iv_off = n;
 
-    return( 0 );
+end:
+    if ( ret != 0 )
+        mbedtls_zeroize( output, length );
```

Comment:
Sorry for not bringing this up earlier, but this probably does not do what it is intended to do: The `output` and `length` values are updated during the operations, so this zeroization will clear the _remaining_, so far unwritten, output buffer, while the part that has already been written is left untouched. Also, the loops decrease `length` at the beginning of the iteration while `output` is increased at the end, so the zeroization will omit one byte at the end.

Reply:
Good catch! I missed that for some reason. I'll fix that.